stl/inc/memory

```diff
@@ -1789,8 +1789,8 @@ protected:
 
 // CLASS TEMPLATE _Ref_count_obj_alloc3
 template <class _Ty, class _Alloc>
-class __declspec(empty_bases) _Ref_count_obj_alloc3 : public _Ebco_base<_Rebind_alloc_t<_Alloc, _Ty>>,
-                                                      public _Ref_count_base {
+class _DECLSPEC_EMPTY_BASES _Ref_count_obj_alloc3 : public _Ebco_base<_Rebind_alloc_t<_Alloc, _Ty>>,

```

Comment:
I came here to scream from the rooftops that allowing differing layout between MSVC/Clang and NVCC for something as common as `shared_ptr` control blocks will inevitably lead to ODR violations of the worst possible kind. I then noticed that `empty_bases` doesn't actually do anything here since each of these classes has at most one base (a specialization of `_Ebco_base`) that is potentially empty (See https://godbolt.org/z/ZA5nLB, which I validated with `/d1reportAllClassLayout`).

I think we can simply remove `__declspec(empty_bases)` on these control block types without altering layout.

Reply:
Ah, excellent! Also, this hasn't shipped yet - all of these control blocks were recently added by `make_shared()` For Arrays. So we can definitely remove `empty_bases`.